upup/pkg/kutil/drain.go

```diff
@@ -0,0 +1,674 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```

Comment:
@kris-nova I have already found two bugs in kubectl.  The code from kubectl may not be robust enough.  Moreover, I am planning on using the API client, in other iterations, as we need to support 1.4.x and 1.5.x.  For instance, 1.4.x kubectl drain fails if you are running a Petset.  Also, this code is removing the local storage of a daemon set, which is not quite correct.  DSs should not be touched.

I would like to work on improving this after this iteration, as this is functioning well currently.